core/src/rpc/format/msgpack/mod.rs

```diff
@@ -0,0 +1,26 @@
+mod convert;
+
+use crate::rpc::format::ResTrait;
+use crate::rpc::RpcError;
+pub use convert::Pack;
+
+use crate::rpc::request::Request;
+use crate::sql::Value;
+
+pub fn req(val: Vec<u8>) -> Result<Request, RpcError> {
+	rmpv::decode::read_value(&mut val.as_slice())
+		.map_err(|_| RpcError::ParseError)
+		.map(Pack)?
+		.try_into()
+}
+
+pub fn res(res: impl ResTrait) -> Result<Vec<u8>, RpcError> {
+	// Convert the response into a value
+	let val: Value = res.into();
+	let val: Pack = val.try_into()?;
+	// Create a new vector for encoding output
+	let mut res = Vec::new();
```

Comment:
Is there a way to determine size? Because you might be able to save allocations and fragmenting if so.

Reply:
I don't know of any simple way to do it. This is the way that it was done before so there won't be any regression. There might be possibility for an optimisation later by analysing the Value and use heuristics to estimate.

Reply:
Heads up that we'll likely be dropping MessagePack anyways...

Reply:
the same thing is done with CBOR though, we could add a method on a sql value or cbor value, which recursively estimates the serialisation length

Reply:
We know that we are always serialising an array of objects at the very least, so even if we pre-allocate something greater than 0, that might be more optimum than just `Vec::new()`. We can do this in a future PR when we drop MessagePack for the next version!